### PR TITLE
PYTHON-3772 add types to change_stream.py

### PR DESCRIPTION
**Pull request by sleepyStick**
https://jira.mongodb.org/browse/PYTHON-3772

### CODE REVIEW

File: pymongo/change_stream.py [hunk 1]
```diff
@@ -226,7 +238,7 @@ def _process_result(self, result, sock_info):
                         "response : {!r}".format(result)
                     )
 
-    def _run_aggregation_cmd(self, session, explicit_session):
+    def _run_aggregation_cmd(self, session: ClientSession, explicit_session: bool) -> CommandCursor:
```
Comment (ShaneHarvey) [inline pymongo/change_stream.py:241]: Should be `Optional[ClientSession]`.

File: pymongo/change_stream.py [hunk 2]
```diff
@@ -246,9 +258,10 @@ def _run_aggregation_cmd(self, session, explicit_session):
 
     def _create_cursor(self):
         with self._client._tmp_session(self._session, close=False) as s:
+            assert s
```
Comment (ShaneHarvey) [inline pymongo/change_stream.py:261]: This assert can fail in practice, let's remove it.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/change_stream.py (hunk 2 above)
Comment (sleepyStick) [reply pymongo/change_stream.py]: Done.

File: pymongo/change_stream.py (hunk 1 above)
Comment (sleepyStick) [reply pymongo/change_stream.py]: Okay, thanks

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Pull request merged**